pkg/config/remote_config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// supported remote config provider schemes
+const (
+	HTTP  = "http"
+	HTTPS = "https"
+	// TODO: add s3, gcs, blob, and git providers backed by go-fsimpl
+)
+
+// RemoteOpts struct contains agent remote config options
+type RemoteOpts struct {
+	URL              *url.URL
```

Comment:
minor DX nit: it's a little unexpected that the user is not supposed to supply URL, but can supply the other field. I'd say make it unexported and keep HTTPClientConfig exported.

Reply:
yeah, that irked me a bit but seemed like the lesser evil compared to having to pass url separately down to the providers. I think un-exporting it is a good solution for now.